src/main/java/com/iexec/sms/secret/web3/Web3SecretRepository.java

```diff
@@ -16,13 +16,14 @@
 
 package com.iexec.sms.secret.web3;
 
-
 import org.springframework.data.repository.CrudRepository;
 
 import java.util.Optional;
 
 public interface Web3SecretRepository extends CrudRepository<Web3Secret, String> {
+    Optional<Web3Secret> findByHeader(Web3SecretHeader header);
 
-    Optional<Web3Secret> findWeb3SecretByAddress(String secretAddress);
-
+    default Optional<Web3Secret> find(String secretAddress) {
+        return findByHeader(new Web3SecretHeader(secretAddress));
```

Comment:
```suggestion
        return findById(new Web3SecretHeader(secretAddress));
```
Extend the interface with `CrudRepository<Web3Secret, Web3SecretHeader>`

Reply:
Good point! Please see https://github.com/iExecBlockchainComputing/iexec-sms/pull/140/commits/9e0e25ddf1284532e8dffd3da038f5333a85a186.